pystac/stac_io.py

```diff
@@ -101,7 +101,7 @@ def json_loads(self, txt: str, *args: Any, **kwargs: Any) -> Dict[str, Any]:
             result = json.loads(txt)
         return result
 
-    def json_dumps(self, json_dict: Dict[str, Any], *args: Any, **kwargs: Any) -> str:
```

Comment:
I believe these are intentional to allow for overriding classes to user arbitrary arguments? cc @matthewhanson

Reply:
That's right. We have these here intentionally in case someone wants to overwrite `json_dumps` with an implementation that needs to pass other arguments to the JSON parser. The same goes for `save_json`, since it calls `json_dumps` using those additional arguments.

Reply:
Yes, these are overridden in Client.

The **kwargs in `generate_subcatalogs` is not used in a subclass that I know of.

Reply:
Why not just pass the arbitrary arguments by default? Then the user won't have to override the implementation if they're using the built-in `json` module or `orjson`.

Reply:
I've moved the relevant commit to https://github.com/stac-utils/pystac/pull/542.